zilliqa/src/consensus.rs

```diff
@@ -563,7 +563,9 @@ impl Consensus {
                     self.new_view_message_cache.as_ref()
                 {
                     if new_view.view == self.get_view()? {
-                        return Ok(self.new_view_message_cache.clone());
+                        // When re-sending new view messages we broadcast them, rather than only sending them to the
+                        // view leader. This speeds up network recovery when many nodes have different high QCs.
+                        return Ok(Some((None, ExternalMessage::NewView(new_view.clone()))));
```

Comment:
We could get away with just caching the `NewView` rather than the `(leader, ExternalMessage)` which would simplify things. Im happy to do as a clean up later